Sources/NautilusTelemetry/Tracing/Span.swift

```diff
@@ -7,6 +7,21 @@
 
 import Foundation
 
+// MARK: - Link
+
+/// Minimal subset of OTLP.SpanLink with relationship concept
+public struct Link {
+	public enum Relationship {
+		case parent
+		case child
+		case undefined
```

Comment:
Do we want the `undefined` case? It looks like it currently produces a "partial" link (one without any attributes).

Reply:
I'm going to keep it for now -- there are potentially other types of relationship (siblings in the same batch for example), but I don't want to go overboard in modeling.